### PR TITLE
Add ServiceAgents model

### PR DESCRIPTION
**Pull request by chrisgilmerproj**
## Description

Related to https://github.com/transcom/mymove/pull/872

When a TSP accepts a move they must submit info for a Origin and Destination shipping agent.  Without this info they cannot accept since we're prioritizing contact info in this phase. 

## Reviewer Notes

Please help me make sure my foreign key work is correct.

## Code Review Verification Steps

* [x] Any migrations/schema changes also update the diagram in docs/schema/dp3.sqs
* [x] Request review from a member of a different team.
* [x] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/n/projects/2181745/stories/159503832) for this change

### CODE REVIEW

File: pkg/models/shipping_agent.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	// AgentTypeORIGIN capture enum value "ORIGIN"
+	AgentTypeORIGIN AgentType = "ORIGIN"
+	// AgentTypeDESTINATION capture enum value "DESTINATION"
+	AgentTypeDESTINATION AgentType = "DESTINATION"
+)
+
+// ShippingAgent represents an assigned agent for a shipment
+type ShippingAgent struct {
+	ID          uuid.UUID `json:"id" db:"id"`
+	ShipmentID  uuid.UUID `json:"shipment_id" db:"shipment_id"`
```
Comment (kilbergr) [inline pkg/models/shipping_agent.go]: Can you explain a bit more about what a shipping agent is? I'm unclear from the PR description and the above comment whether a shipment should be required or not. You have it not required.

**Comment by chrisgilmerproj** [comment]
@kilbergr - happy to add more detail here.  Also, it came to my attention at the end of last week that I could be using a more generic term here so I'll probably push a change shortly. 

A "shipping agent" is the company or group that a TSP uses to actually pickup, move, or unload the stuff being shipped.  They can be the TSP or they can be sub contractors or agents that the work is divided up with.  Before accepting a move a TSP needs to provide both the Origin and the Destination shipping agent.  However, in practice often the Origin shipping agent is the only one put in the system, the Destination agent is the TSP and when an actual Destination agent is decided upon the TSP will go in and update that record.

Talking with SBD I learned that the more generic term is "Service Agent".  Because it's not always just about shipping but could include more steps like hiring out a group that does the packing or un-packing to be different than the ones loading and unloading.

Here's a screen shot I got from SBD's demo:

screen shot 2018-08-10 at 2 07 11 pm

**Comment by chrisgilmerproj** [comment]
Alright @kilbergr - would you take another look? I think I have this updated the way I'd like it.

File: migrations/20180810112708_create_service_agents.up.fizz [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+	t.Column("shipment_id", "uuid", {})
+	t.Column("role", "text", {})
+	t.Column("point_of_contact", "text", {})
+	t.Column("email", "text", {"null": true})
+	t.Column("phone_number", "text", {"null": true})
+	t.Column("fax_number", "text", {"null": true})
+	t.Column("email_is_preferred", "bool", {"null": true})
+	t.Column("phone_is_preferred", "bool", {"null": true})
+	t.Column("notes", "text", {"null": true})
+	t.ForeignKey("shipment_id", {"shipments": ["id"]}, {})
```
Comment (kilbergr) [inline migrations/20180810112708_create_service_agents.up.fizz]: Do you need Company as a column here?

**Pull request merged**